src/common/charts/LegendChart.jsx

```diff
@@ -70,7 +72,7 @@ module.exports = React.createClass({
       <div style={{'width': props.width, 'height': props.height}} >
         {this._renderTitle()}
         {this._renderLegend()}
-        <svg viewBox={props.viewBox} width={props.width - props.sideOffset} height={props.height}>{props.children}</svg>
+        <svg viewBox={this.getViewBox()} width={this.getOuterDimensions().width - props.sideOffset} height={props.height}>{props.children}</svg>
```

Comment:
I think `width={this.getOuterDimensions().width}` will result in having a static `px` width set on the svg element instead of a `%` that gets passed in. I see this is a complication because of the arithmetic that needs to be done to account for the legend's size. I think the layout system needs to be changed to use some sort of table layout or floats so that the svg can be `width=100%` while still allowing for room for the legend.